File: update_pricing.js
const fs = require('fs');
let content = fs.readFileSync('src/pages/Pricing.tsx', 'utf8');

const importReplacement = `import React from 'react';
import { useNavigate, useSearchParams } from 'react-router-dom';
import { Card, CardContent, CardHeader, CardTitle } from '@/src/components/ui/card';
import { Button } from '@/src/components/ui/button';
import { Check, ChevronDown } from 'lucide-react';
import { useDocumentTitle } from '../hooks/useDocumentTitle';
import type { ModuleType } from '../types';

const availableModules: ModuleType[] = [
  'Retail POS',
  'Manufacturing ERP',
  'Education',
  'Health Care',
  'Hotel/Restaurant',
  'Transport Management',
  'Daily Services',
  'Agri Management'
];`;

content = content.replace(/import React.*useDocumentTitle';/s, importReplacement);

const componentStartReplace = `export default function Pricing() {
  useDocumentTitle('BahiBox | Pricing');
  const navigate = useNavigate();
  const [searchParams, setSearchParams] = useSearchParams();
  const moduleName = searchParams.get('module') || 'Retail POS';

  const handleSelectPlan = (plan: string) => {
    navigate(\`/checkout?module=\${encodeURIComponent(moduleName)}&plan=\${encodeURIComponent(plan)}\`);
  };`;

content = content.replace(/export default function Pricing\(\) \{.*?navigate\(\`\/checkout\?module=\\\$\{\encodeURIComponent\(moduleName\)\}&plan=\\\$\{\encodeURIComponent\(plan\)\}\`\);\s*\};/s, componentStartReplace);

const headingReplace = `<h1 className="text-3xl md:text-4xl font-extrabold text-slate-900 mb-4">
            Pricing for 
            <div className="inline-block relative ml-2">
              <select 
                className="appearance-none bg-transparent border-b-2 border-primary text-primary pr-8 pl-2 py-1 outline-none cursor-pointer hover:bg-slate-100 rounded-t transition-colors"
                value={moduleName}
                onChange={(e) => setSearchParams({ module: e.target.value })}
              >
                {availableModules.map(mod => (
                  <option key={mod} value={mod}>{mod}</option>
                ))}
              </select>
              <ChevronDown className="absolute right-2 top-1/2 -translate-y-1/2 h-5 w-5 text-primary pointer-events-none" />
            </div>
          </h1>`;

content = content.replace(/<h1 className="text-3xl md:text-4xl font-extrabold text-slate-900 mb-4">\s*Pricing for \{moduleName\}\s*<\/h1>/s, headingReplace);

fs.writeFileSync('src/pages/Pricing.tsx', content);

File: update_pricing.py
with open('src/pages/Pricing.tsx', 'r') as f:
    content = f.read()

target1 = """function getOfficialModuleName(raw: string, fallback: string = 'Retail POS') {
  if (!raw) return fallback;
  const l = raw.toLowerCase();
  if (l.includes('retail')) return 'Retail POS';
  if (l.includes('manufacturing')) return 'Manufacturing';
  if (l.includes('education')) return 'Education';
  if (l.includes('healthcare') || l.includes('health')) return 'Healthcare';
  if (l.includes('hotel') || l.includes('hospitality') || l.includes('restaurant')) return 'Hotel & Restaurant';
  if (l.includes('transport') || l.includes('logistic')) return 'Transport';
  if (l.includes('service')) return 'Daily Services';
  if (l.includes('agri') || l.includes('farm')) return 'Agriculture';
  return raw;
}
"""

repl1 = ""
if target1 in content:
    content = content.replace(target1, repl1)
else:
    print("target1 not found")

target2 = """  const availableModules = Array.from(new Set(plans.map(p => p.moduleName)));
  const defaultModule = availableModules.includes('Retail POS') ? 'Retail POS' : (availableModules[0] || 'Retail POS');
  const rawModuleName = (searchParams.get('module') || defaultModule) as string;
  const moduleName = getOfficialModuleName(rawModuleName, rawModuleName);"""

repl2 = """  const availableModules = Array.from(new Set(plans.map(p => p.moduleName)));
  const defaultModuleKey = plans.find(p => p.moduleName === 'Retail POS')?.moduleKey || (plans[0]?.moduleKey || 'retail');
  const rawModuleKey = (searchParams.get('module') || defaultModuleKey) as string;
  const moduleName = plans.find(p => p.moduleKey === rawModuleKey)?.moduleName || rawModuleKey;"""

if target2 in content:
    content = content.replace(target2, repl2)
else:
    print("target2 not found")


target3 = """                <Select value={moduleName} onValueChange={(val) => setSearchParams({ module: val })}>
                  <SelectTrigger className="w-full h-12 text-lg font-medium bg-white dark:bg-slate-950">
                    <SelectValue placeholder="Select Module" />
                  </SelectTrigger>
                  <SelectContent>
                    {availableModules.map(mod => (
                      <SelectItem key={mod} value={mod}>{getOfficialModuleName(mod, mod)}</SelectItem>
                    ))}
                  </SelectContent>
                </Select>"""

repl3 = """                <Select value={rawModuleKey} onValueChange={(val) => setSearchParams({ module: val })}>
                  <SelectTrigger className="w-full h-12 text-lg font-medium bg-white dark:bg-slate-950">
                    <SelectValue placeholder="Select Module" />
                  </SelectTrigger>
                  <SelectContent>
                    {availableModules.map(mod => {
                      const key = plans.find(p => p.moduleName === mod)?.moduleKey || mod;
                      return <SelectItem key={key} value={key}>{mod}</SelectItem>;
                    })}
                  </SelectContent>
                </Select>"""

if target3 in content:
    content = content.replace(target3, repl3)
else:
    print("target3 not found")

with open('src/pages/Pricing.tsx', 'w') as f:
    f.write(content)
print("done")
